docarray/utils/reduce.py

```diff
@@ -0,0 +1,164 @@
+from docarray import DocumentArray
+from typing import List, Optional, Dict, _GenericAlias  # type: ignore
+from docarray.base_document import BaseDocument
+
+
+def _types_analysis(doc: 'BaseDocument') -> List[List[str]]:
+    simple_non_empty_fields: List[str] = []
+    list_fields: List[str] = []
+    set_fields: List[str] = []
+    nested_docs_fields: List[str] = []
+    nested_docarray_fields: List[str] = []
+
+    for field_name, field in doc.__fields__.items():
+        field_type = field.outer_type_
+        if not isinstance(field_type, _GenericAlias) and issubclass(
+            field_type, DocumentArray
+        ):
+            nested_docarray_fields.append(field_name)
+        elif isinstance(field_type, _GenericAlias) and field_type.__origin__ is list:
+            list_fields.append(field_name)
+        elif isinstance(field_type, _GenericAlias) and field_type.__origin__ is set:
+            set_fields.append(field_name)
+        v = getattr(doc, field_name)
+        if v:
+            if isinstance(v, BaseDocument):
+                nested_docs_fields.append(field_name)
+            else:
+                simple_non_empty_fields.append(field_name)
+    return [
+        simple_non_empty_fields,
+        list_fields,
+        set_fields,
+        nested_docarray_fields,
+        nested_docs_fields,
+    ]
+
+
+def reduce_docs(doc1: 'BaseDocument', doc2: 'BaseDocument') -> 'BaseDocument':
```

Comment:
Would it make sense to generalize this to take in an number or `*docs` and do things in a loop? Or is it overkill?

Reply:
I think is overkill, it has been moved to `update`